scene/resources/dynamic_font.cpp

```diff
@@ -303,14 +326,142 @@ void DynamicFontAtSize::set_texture_flags(uint32_t p_flags) {
 	}
 }
 
+bool DynamicFontAtSize::shape_run(Vector<Ref<ClusterInfo> > &p_visual_order, Vector<Ref<ClusterInfo> > &p_logical_order, const Vector<UChar> &p_utf16_text, int p_start, int p_length, TextDirection p_direction, const Vector<hb_feature_t> &p_features, const String &p_script, const String &p_lang) const {
+
+	hb_buffer_t *buffer = hb_buffer_create();
```

Comment:
You can re-use the `hb_buffer_t` instead of creating one from scratch for each run. `hb_buffer_clear_contents()` should be called before re-using the buffer.